library/src/main/java/com/novoda/downloadmanager/DownloadFileIdCreator.java

```diff
@@ -7,6 +7,6 @@ private DownloadFileIdCreator() {
     }
 
     public static DownloadFileId createFrom(String id) {
-        return new LiteDownloadFileId(id.hashCode());
+        return new LiteDownloadFileId(id);
```

Comment:
Don't these need to be unique? Does this change enforce that?

Reply:
I don't believe these need to be unique for the `download-manager`, we don't really deal with Ids. The only time this is useful is when retrieving a `DownloadFileStatus`. If a client duplicates the ids then they will only receive the first matching `DownloadFileStatus`. Hashcode wouldn't have enforced uniqueness and broke the app because we were using the creator in multiple places for an id that was meant to remain constant.

Reply:
You're right, I was thinking of the `filename`